engine/access/rest/websockets/data_providers/transaction_statuses_provider.go

```diff
@@ -0,0 +1,173 @@
+	"github.com/onflow/flow-go/engine/access/rest/websockets/models"
+	"github.com/onflow/flow-go/engine/access/subscription"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/counters"
+
+	"github.com/onflow/flow/protobuf/go/flow/entities"
+)
+
+// transactionStatusesArguments contains the arguments required for subscribing to transaction statuses
+type transactionStatusesArguments struct {
+	TxID             flow.Identifier // ID of the transaction to monitor.
+	StartBlockID     flow.Identifier // ID of the block to start subscription from
+	StartBlockHeight uint64          // Height of the block to start subscription from
+}
+
+type TransactionStatusesDataProvider struct {
+	*baseDataProvider
+
+	logger zerolog.Logger
+	api    access.API
+}
+
+var _ DataProvider = (*TransactionStatusesDataProvider)(nil)
+
+func NewTransactionStatusesDataProvider(
+	ctx context.Context,
+	logger zerolog.Logger,
+	api access.API,
+	topic string,
+	arguments models.Arguments,
+	send chan<- interface{},
+) (*TransactionStatusesDataProvider, error) {
+	p := &TransactionStatusesDataProvider{
+		logger: logger.With().Str("component", "transaction-statuses-data-provider").Logger(),
+		api:    api,
+	}
+
+	// Initialize arguments passed to the provider.
+	txStatusesArgs, err := parseTransactionStatusesArguments(arguments)
+	if err != nil {
+		return nil, fmt.Errorf("invalid arguments for tx statuses data provider: %w", err)
+	}
+
+	subCtx, cancel := context.WithCancel(ctx)
+
+	p.baseDataProvider = newBaseDataProvider(
+		topic,
+		cancel,
+		send,
+		p.createSubscription(subCtx, txStatusesArgs), // Set up a subscription to tx statuses based on arguments.
+	)
+
+	return p, nil
+}
+
+// Run starts processing the subscription for events and handles responses.
+//
+// No errors are expected during normal operations.
+func (p *TransactionStatusesDataProvider) Run() error {
+	return subscription.HandleSubscription(p.subscription, p.handleResponse())
+}
+
+// createSubscription creates a new subscription using the specified input arguments.
+func (p *TransactionStatusesDataProvider) createSubscription(
+	ctx context.Context,
+	args transactionStatusesArguments,
+) subscription.Subscription {
+	if args.StartBlockID != flow.ZeroID {
+		return p.api.SubscribeTransactionStatusesFromStartBlockID(ctx, args.TxID, args.StartBlockID, entities.EventEncodingVersion_JSON_CDC_V0)
+	}
+
+	if args.StartBlockHeight != request.EmptyHeight {
+		return p.api.SubscribeTransactionStatusesFromStartHeight(ctx, args.TxID, args.StartBlockHeight, entities.EventEncodingVersion_JSON_CDC_V0)
+	}
+
+	return p.api.SubscribeTransactionStatusesFromLatest(ctx, args.TxID, entities.EventEncodingVersion_JSON_CDC_V0)
+}
+
+// handleResponse processes an account statuses and sends the formatted response.
+//
+// No errors are expected during normal operations.
+func (p *TransactionStatusesDataProvider) handleResponse() func(txResults []*access.TransactionResult) error {
```

Comment:
Can we use  generic `HandleResponse` from `subscription` in `Run()` and get rid of  `handleResponse()`?